plugins/network-elements/nuage-vsp/src/com/cloud/agent/api/guru/TrashNetworkVspCommand.java

```diff
@@ -48,13 +53,93 @@ public boolean isL3Network() {
         return _isL3Network;
     }
 
+    public boolean isSharedNetwork() {
+        return _isSharedNetwork;
+    }
+
     public String getVpcUuid() {
         return _vpcUuid;
     }
 
+    public String getDomainTemplateName() {
+        return _domainTemplateName;
+    }
+
+    public static class Builder implements CmdBuilder<TrashNetworkVspCommand> {
+        private String _domainUuid;
+        private String _networkUuid;
+        private boolean _isL3Network;
+        private boolean _isSharedNetwork;
+        private String _vpcUuid;
+        private String _domainTemplateName;
+
+        public Builder domainUuid(String domainUuid) {
+            this._domainUuid = domainUuid;
+            return this;
+        }
+
+        public Builder networkUuid(String networkUuid) {
+            this._networkUuid = networkUuid;
+            return this;
+        }
+
+        public Builder isL3Network(boolean isL3Network) {
+            this._isL3Network = isL3Network;
+            return this;
+        }
+
+        public Builder isSharedNetwork(boolean isSharedNetwork) {
+            this._isSharedNetwork = isSharedNetwork;
+            return this;
+        }
+
+        public Builder vpcUuid(String vpcUuid) {
+            this._vpcUuid = vpcUuid;
+            return this;
+        }
+
+        public Builder domainTemplateName(String domainTemplateName) {
+            this._domainTemplateName = domainTemplateName;
+            return this;
+        }
+
+        @Override
+        public TrashNetworkVspCommand build() {
+            return new TrashNetworkVspCommand(_domainUuid, _networkUuid, _isL3Network, _isSharedNetwork, _vpcUuid, _domainTemplateName);
+        }
+    }
+
     @Override
     public boolean executeInSequence() {
         return false;
     }
 
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+
+        TrashNetworkVspCommand that = (TrashNetworkVspCommand) o;
+
+        if (_isL3Network != that._isL3Network) return false;
+        if (_isSharedNetwork != that._isSharedNetwork) return false;
+        if (_domainTemplateName != null ? !_domainTemplateName.equals(that._domainTemplateName) : that._domainTemplateName != null)
+            return false;
+        if (_domainUuid != null ? !_domainUuid.equals(that._domainUuid) : that._domainUuid != null) return false;
+        if (_networkUuid != null ? !_networkUuid.equals(that._networkUuid) : that._networkUuid != null) return false;
+        if (_vpcUuid != null ? !_vpcUuid.equals(that._vpcUuid) : that._vpcUuid != null) return false;
+
+        return true;
+    }
+
+    @Override
+    public int hashCode() {
```

Comment:
This ``hashCode`` implementation should include attributes from the ``Command`` super class.